src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs

```diff
@@ -4,26 +4,31 @@
 // </copyright>
 //---------------------------------------------------------------------
 
-using Microsoft.OData.Edm.Vocabularies;
-
 namespace Microsoft.OData.Edm.Csdl.Parsing.Ast
 {
     /// <summary>
-    /// Represents a CSDL function return type.
+    /// Represents a CSDL operation return type.
     /// </summary>
-    internal class CsdlOperationReturnType : CsdlElement, IEdmVocabularyAnnotatable
+    internal class CsdlOperationReturnType : CsdlElement
     {
-        private readonly CsdlTypeReference returnType;
-
         public CsdlOperationReturnType(CsdlTypeReference returnType, CsdlLocation location)
             : base(location)
         {
-            this.returnType = returnType;
+            this.ReturnType = returnType;
+        }
+
+        public CsdlTypeReference ReturnType { get; }
+
+        public override void AddAnnotation(CsdlDirectValueAnnotation annotation)
+        {
+            base.AddAnnotation(annotation);
+            ReturnType.AddAnnotation(annotation);

```

Comment:
Is this right? Should we be annotating both the OperationReturnType and the CsdlTypeReference?  I would have expected that we only annotate the CsdlOperationReturnType...

Reply:
I have to do like this, and I double check that's ok. 
The reason is that:

1. Only the "CsdlTypeReference" will be used to constructor the "CsdlSemanticsOperation....ReturnType"'.

2. CsdlOperationReturnType is only a middle wrapper class, it's not used in later to constructor the "CsdlSemanticsOperation....ReturnType"'.